File: Python/Island Perimeter.py
class Solution(object):
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        counter = 0
        self.row = len(grid)
        self.col = len(grid[0])
        for line in grid:
            line.append(0)
            line.insert(0,0)
        grid.insert(0, [0] * (self.col + 2))
        grid.append([0] * (self.col + 2))
        for i in xrange(1, self.row + 1):
            for j in xrange(1, self.col + 1):
                if grid[i][j] == 1:
                    counter += self.count(i, j, grid)
        return counter
        
        
    
    def count(self, i, j, grid):
        counter = 0
        if grid[i - 1][j] == 0:
            counter += 1
        if grid[i + 1][j] == 0:
            counter += 1
        if grid[i][j + 1] == 0:
            counter += 1
        if grid[i][j - 1] == 0:
            counter += 1
        return counter
            
        
        

File: cpp/Island Perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        unsigned int width = grid[0].size();
        vector<int> first_line = vector<int>(width, 0);
        vector<int> last_line = vector<int>(width, 0);
        grid.push_back(last_line);
        grid.insert(grid.begin(), first_line);

        for(int i = 0; i < grid.size(); i++) {
            grid[i].push_back(0);
            grid[i].insert(grid[i].begin(), 0);
        }

        int counter = 0;
        for (int row = 1; row < grid.size() -1; ++row) {
            for (int col = 1; col < width + 1; ++col) {
                if (grid[row][col]) {
                    if (!grid[row - 1][col])
                        counter++;
                    if (!grid[row + 1][col])
                        counter++;
                    if (!grid[row][col - 1])
                        counter++;
                    if (!grid[row][col + 1])
                        counter++;
                }
            }
        }
        return counter;
    }
};
